libraries-3/src/test/java/com/baeldung/jnats/NatsClientLiveTest.java

```diff
@@ -1,115 +1,123 @@
 package com.baeldung.jnats;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertNull;
+import io.nats.client.Message;
+import io.nats.client.Subscription;
+import org.junit.Test;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
 
-import org.junit.Test;
-
-import io.nats.client.Message;
-import io.nats.client.SyncSubscription;
+import static org.junit.Assert.*;
 
 public class NatsClientLiveTest {
 
+    private NatsClient connectClient() {
+        return new NatsClient();
+    }
+
     @Test
-    public void givenMessageExchange_MessagesReceived() throws Exception {
+    public void messageExchangeViaPublishSubscribe_MessagesReceived() throws Exception {
```

Comment:
will fix

Reply:
This hasn't been addressed, the syntax is either:
1. givenSomethingBeforeTheWhen_whenTheActionIsTaken_thenTheExpectedResult
2. whenTheActionIsTaken_thenTheExpectedResult

Reply:
No, this is not correct because:
1. There is no _whenXXX
2. There is no _thenXXX
3. _MessageReceived is not camelCase
4. givenXXX_thenXXX is not a valid pattern, it's always given_when_then or when_then, never just given_then